libbeat/autodiscover/autodiscover.go

```diff
@@ -127,23 +127,37 @@ func (a *Autodiscover) Start() {
 
 func (a *Autodiscover) worker() {
 	var updated, retry bool
-
+	ticker := time.NewTicker(retryPeriod)
 	for {
-		select {
-		case event := <-a.listener.Events():
-			// This will happen on Stop:
-			if event == nil {
-				return
-			}
+		// accumulate updates for every event that comes in so that we dont overwhelm
+		// the reloading engine with too many updates
```

Comment:
@jsoriano this is related to the other one you were working on at https://github.com/elastic/beats/pull/29048, right?

Reply:
Well, #29048 can certainly reduce the number of configurations reloaded, but it doesn't do anything to reuse hash calculations. It can be a good idea to calculate them only once for each config, but I think this can be done as a different change.